airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultCheckConnectionWorker.java

```diff
@@ -104,25 +95,30 @@ public ConnectorJobOutput run(final StandardCheckConnectionInput input, final Pa
             });
       }
 
-      if (status.isPresent() && exitCode == 0) {
+      final Optional<FailureReason> failureReason = WorkerUtils.getJobFailureReasonFromMessages(OutputType.CHECK_CONNECTION, messagesByType);
+      failureReason.ifPresent(jobOutput::setFailureReason);
+
+      final int exitCode = process.exitValue();
+      if (exitCode != 0) {
+        LOGGER.warn("Check connection job subprocess finished with exit code {}", exitCode);
+      }
+
+      if (connectionStatus.isPresent()) {
         final StandardCheckConnectionOutput output = new StandardCheckConnectionOutput()
-            .withStatus(Enums.convertTo(status.get().getStatus(), Status.class))
-            .withMessage(status.get().getMessage());
-
-        LOGGER.debug("Check connection job subprocess finished with exit code {}", exitCode);
-        LOGGER.debug("Check connection job received output: {}", output);
-        LineGobbler.endSection("CHECK");
-        return new ConnectorJobOutput().withOutputType(OutputType.CHECK_CONNECTION).withCheckConnection(output);
-      } else {
-        final String message = String.format("Error checking connection, status: %s, exit code: %d", status, exitCode);
-        LOGGER.error(message);
-
-        return WorkerUtils.getJobFailureOutputOrThrow(OutputType.CHECK_CONNECTION, messagesByType, message);
+            .withStatus(Enums.convertTo(connectionStatus.get().getStatus(), Status.class))
+            .withMessage(connectionStatus.get().getMessage());
+        LOGGER.info("Check connection job received output: {}", output);
+        jobOutput.setCheckConnection(output);
```

Comment:
So we determine success by checking the `jobOutput` for a `checkConnection`, which will only be set if there is a connectionStatus.

We return the jobOutput either way, but if it's successful we can `getCheckConnection`, otherwise we can't and we need to check the FailureReason to figure out why

Reply:
Do we need to worry at all about any places where `worker.run()` might interpret receiving a JobOutput with no checkConnection but with a FailureReason as a success, since we used to always return a jobOutput with a checkConnection, or throw ?